
version: '3.7'

services:
  pihole:
    image: pihole/pihole:latest
    hostname: "{{pihole_app_name}}"
    user: "root:root"
    privileged: true
    ports:
      - "{{pihole_ftl_dns_port}}:{{pihole_ftl_dns_port}}/udp"
      - "8085:{{pihole_web_port}}"
    restart: always
    volumes:
      - /etc/pihole:/etc/pihole
      - "/home/{{ansible_user}}/stacks/lighttpd/external.conf:/etc/lighttpd/external.conf"
      - dnsmasq:/etc/dnsmasq.d
    environment:
      - "TZ=Europe/London"
      - "DNS1=1.1.1.1"
      - "DNS2=8.8.4.4"
      - "REV_SERVER=true"
      - "REV_SERVER_CIDR=192.168.1.0/24" #Update these fields to match your environment
      - "REV_SERVER_TARGET={{REV_SERVER_TARGET}}"
      - "REV_SERVER_DOMAIN={{app_domain_name}}"
      - "WEBPASSWORD={{pihole_web_password}}"
    cap_add:
        - SYS_PTRACE
    networks:
      - bridge
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.pihole-http.rule=Host(`{{pihole_app_name}}.{{app_domain_name}}`)"
      - "traefik.http.routers.pihole-http.entrypoints={{traefik_network_name}}"
      - "traefik.http.middlewares.add-admin-prefix.addprefix.prefix=/admin"
      - "traefik.http.routers.pihole-http.middlewares=add-admin-prefix@docker"
      - "traefik.http.routers.pihole-http.service=pihole-http-svc"
      - "traefik.http.services.pihole-http-svc.loadbalancer.server.port={{pihole_web_port}}"
      - "traefik.http.services.pihole-http-svc.loadbalancer.passhostheader=true"

volumes:
  dnsmasq:

networks:
  {{traefik_network_name}}:
    external: true
    name: {{traefik_network_name}}